src/components/Avatar/AvatarStack/AvatarStack.tsx

```diff
@@ -30,27 +30,35 @@ const AvatarStack = React.forwardRef<HTMLDivElement, Props & TestProps & DivProp
     },
     ref
   ) => {
-    const clampedMax = maxAvatars < 1 ? 1 : maxAvatars;
+    if (maxAvatars < 1) {
+      throw new Error('maxAvatars prop must be greater than 0');
+    }
 
     const children = React.Children.toArray(childrenProp);
 
-    const extraAvatars = children.length > clampedMax ? children.length - clampedMax : 0;
+    const extraAvatars = children.length > maxAvatars ? children.length - maxAvatars : 0;
+
+    const renderContent = useCallback(
+      () =>
+        children.slice(0, children.length - extraAvatars).map((child, index) => {
+          return (
+            <div key={index} css={avatarWrapperStyle({ zIndex: children.length - index, size })}>
+              {child}
+            </div>
+          );
+        }),
+      [children, extraAvatars, size]
+    );
 
     return (
       <div
         ref={ref}
         data-testid={generateTestDataId('avatarstack', dataTestId)}
-        style={{ display: 'flex' }}
+        css={avatarStackStyle({ size })}
       >
-        {children.slice(0, children.length - extraAvatars).map((child, index) => {
-          return (
-            <div key={index} css={avatarStackStyle({ zIndex: children.length - index, size })}>
-              {child}
-            </div>
-          );
-        })}
+        {renderContent()}
         {extraAvatars ? (
-          <div css={avatarStackStyle({ zIndex: 0, size })}>
+          <div css={avatarWrapperStyle({ zIndex: 0, size })}>
```

Comment:
zindex is no longer needed

Reply:
why not?

Reply:
![image](https://user-images.githubusercontent.com/28539607/148913684-52dced60-a83c-46c2-8277-95daeb16e516.png)
My mistake. I thought the overlap was the other way as shown in the picture above. Double checked it.